src/compas/geometry/_core/basic.py

```diff
@@ -110,7 +110,8 @@ def allclose(l1, l2, tol=1e-05):
     .. [1] https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html
 
     """
-    if any(fabs(a - b) > tol for a, b in zip(l1, l2)):
+
+    if any(not allclose(a, b, tol) if hasattr(a, '__iter__') else fabs(a - b) > tol for a, b in zip(l1, l2)):
```

Comment:
?

Reply:
which extreme boundary case are we accommodating here?

Reply:
is not a boundary case at all, this recursively compares if the element is an iterable, ie `list of list of floats` can be compared. This is how `np.allclose` behaves. The reasons for replicating the behavior is not just consistency, but actually the need to remove all `numpy` from a lot of unit tests that were using `np.allclose` to check results (but were not using `numpy` at all other than that).

Reply:
coolio